### PR TITLE
Add rest module

### PR DESCRIPTION
**Pull request by mrnustik**
Resolves #65
Adds rest module with default configuration.

Wait for #76 and #77 to be merged,.

### CODE REVIEW

File: formulaoneteam-rest/src/test/java/cz/muni/fi/pa165/rest/controllers/MainControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            .build();
+
+    @Test
+    public void onGetRequest_statusIsOk() throws Exception {
+        //Then
+        mockMvc.perform(get("/"))
+                .andExpect(status().isOk());
+    }
+
+    @Test
```
Comment (elderanakain) [inline formulaoneteam-rest/src/test/java/cz/muni/fi/pa165/rest/controllers/MainControllerTest.java]: Please, use JUnit's @Test annotation to be compliant with other tests.

File: formulaoneteam-rest/src/test/java/cz/muni/fi/pa165/rest/controllers/MainControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+import static org.springframework.test.web.servlet.setup.MockMvcBuilders.standaloneSetup;
+
+/**
+ * @author mrnda (Michal Mrnuštík)
+ */
+
+@WebAppConfiguration
+@ContextConfiguration(classes = {RootWebContext.class})
+public class MainControllerTest extends AbstractTestNGSpringContextTests {
```
Comment (elderanakain) [inline formulaoneteam-rest/src/test/java/cz/muni/fi/pa165/rest/controllers/MainControllerTest.java]: Could you also add tests for other classes?
You can make them as simple as possible, so, we can be sure that they are fine during CI check. Also, it wouldn't drop the code coverage.

File: formulaoneteam-rest/src/main/webapp/WEB-INF/web.xml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<!DOCTYPE web-app PUBLIC
+        "-//Sun Microsystems, Inc.//DTD Web Application 2.3//EN"
+        "http://java.sun.com/dtd/web-app_2_3.dtd" >
+
+<web-app>
+    <display-name>Rest interface</display-name>
+
+</web-app>
```
Comment (elderanakain) [inline formulaoneteam-rest/src/main/webapp/WEB-INF/web.xml]: Please, add an empty line at the end of the file.

File: formulaoneteam-rest/src/main/java/cz/muni/fi/pa165/rest/RootWebContext.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+/**
+ * @author mrnda (Michal Mrnuštík)
+ */
+
+@EnableWebMvc
+@Configuration
+@Import({ServiceConfiguration.class, FormulaOneTeamWithSampleDataConfiguration.class})
+@ComponentScan(basePackages = {"cz.fi.muni.pa165.rest.controllers"})
```
Comment (elderanakain) [inline formulaoneteam-rest/src/main/java/cz/muni/fi/pa165/rest/RootWebContext.java]: Please, fix the `basePackage` it is not valid one.

File: formulaoneteam-rest/pom.xml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.glassfish</groupId>
+            <artifactId>javax.el</artifactId>
+            <scope>test</scope> <!-- conflicts with tomcat libraries, but needed for tests to run -->
+        </dependency>
+        
+    </dependencies>
+</project>
```
Comment (elderanakain) [inline formulaoneteam-rest/pom.xml]: Please, add an empty line at the end of the file.

Comment (mrnustik) [reply formulaoneteam-rest/pom.xml]: Done

File: formulaoneteam-rest/src/main/java/cz/muni/fi/pa165/rest/RootWebContext.java (hunk 4 above)
Comment (mrnustik) [reply formulaoneteam-rest/src/main/java/cz/muni/fi/pa165/rest/RootWebContext.java]: Done

File: formulaoneteam-rest/src/main/webapp/WEB-INF/web.xml (hunk 3 above)
Comment (mrnustik) [reply formulaoneteam-rest/src/main/webapp/WEB-INF/web.xml]: Done

File: formulaoneteam-rest/src/test/java/cz/muni/fi/pa165/rest/controllers/MainControllerTest.java (hunk 1 above)
Comment (mrnustik) [reply formulaoneteam-rest/src/test/java/cz/muni/fi/pa165/rest/controllers/MainControllerTest.java]: Done

File: formulaoneteam-rest/src/test/java/cz/muni/fi/pa165/rest/controllers/MainControllerTest.java (hunk 2 above)
Comment (mrnustik) [reply formulaoneteam-rest/src/test/java/cz/muni/fi/pa165/rest/controllers/MainControllerTest.java]: I have added test for the `AllowOriginInterceptor `class. Is that sufficient. I am not sure whether its needed to test also the configuration classes.

**Pull request merged**